### PR TITLE
Updated goccy/go-yaml to v.14.1

### PR DESCRIPTION
**Pull request by XuechunHou**
## Description
Updated `goccy/go-yaml` to `v.14.1` by executing the command `go get -u github.com/goccy/go-yaml@v1.14.1`

## Related issue
b/376658816

## Checklist:
- Unit tests
  - [] Unit tests do not apply.
  - [ ] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [X] Integration tests do not apply.
  - [ ] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [X] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [X] This PR introduces no new features.
  - [ ] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

**Review by braydonk — COMMENTED** [review]
Thanks for doing this. Looks like there is a breaking change in the pretty printing that this PR caught. I'm going to open an issue upstream about this. Our update can be on hold until then.

**Comment by braydonk** [comment]
Opened https://github.com/goccy/go-yaml/issues/534

**Comment by braydonk** [comment]
@XuechunHou The upstream issue has been opened. Can you update the package to `v1.14.1` and update the goldens?

File: integration_test/metadata/integration_metadata.go
Comment (franciscovalentecastro) [inline integration_test/metadata/integration_metadata.go:157]: Please update this comment if this issue (# 313) has been fixed with the current update.

**Review by franciscovalentecastro — APPROVED** [review]
LGTM modulo the integration test pass and addressing comments.

File: transformation_test/transformation_test.go
Comment (franciscovalentecastro) [inline transformation_test/transformation_test.go:572]: I'm curious, why did it has to change from tabs (`\t`) to two spaces ?

Comment (braydonk) [reply transformation_test/transformation_test.go:572]: Using `\t` (tab) where we did is illegal according to the YAML spec. Tabs can't be used as indentation, and putting the `\t` character where we did ended up mixing space and tab indentation. The old version of the YAML library parsed this incorrectly, allowing it to be possible when it should be illegal. This is fixed in this newer version of the library. I decided on two spaces to maintain the original intent of the replacement while still making it legal YAML.

File: integration_test/metadata/integration_metadata.go
Comment (braydonk) [reply integration_test/metadata/integration_metadata.go:157]: Won't be fixed until 1.14.3, which is available if we want to update to that.

**Comment by XuechunHou** [comment]
> LGTM modulo the integration test pass and addressing comments.

Ops Agent and Third party integration tests are failing, though they are already failing at Master head.  Also, these failed tests on github are passing on Louhi.  

These tests mostly failed because "context deadline exceeded"

**Pull request merged**
